azure-pipelines/publishing/github-release.yml

```diff
@@ -9,96 +9,109 @@ pr:
+        config: release
 
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '25'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-agent arm artifacts'
+  - job: deliveryoptimization_plugin_apt
+    steps:
+    - template: ../build/templates/dopapt-native-steps.yml
+      parameters:
+        targetOsArch: 'ubuntu1804-x64'
+        config: release
+    - template: ../build/templates/dopapt-docker-steps.yml
+      parameters:
+        targetOsArch: debian10-arm32
+        dockerImageName: 'jimsonmsft/debian10-arm32:latest'
+        config: release
+    - template: ../build/templates/dopapt-docker-steps.yml
+      parameters:
+        targetOsArch: debian9-arm32
+        dockerImageName: 'jimsonmsft/debian9-arm32:latest'
+        config: release
+    - template: ../build/templates/dopapt-docker-steps.yml
+      parameters:
+        targetOsArch: ubuntu1804-arm64
+        dockerImageName: 'jimsonmsft/ubuntu18.04-arm64:latest'
+        config: release
 
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '23'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-agent x64 artifacts'
-
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '31'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-pluginapt arm artifacts'
-
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: 'specific'
-      project: '7cdf52bf-a6f7-436c-b12f-3f063ef389c4'
-      pipeline: '29'
-      buildVersionToDownload: 'latest'
-      downloadType: 'specific'
-      downloadPath: '/tmp/do_publishing'
-    displayName: 'Get ms-do-pluginapt x64 artifacts'
-
-  - task: ArchiveFiles@2
-    inputs:
-      rootFolderOrFile: '/tmp/do_publishing'
-      includeRootFolder: true
-      archiveType: 'tar'
-      archiveFile: '$(Build.ArtifactStagingDirectory)/artifacts.tar.gz'
-      replaceExistingArchive: true
-
-  - task: GitHubRelease@1
-    inputs:
-      gitHubConnection: 'client2'
-      repositoryName: 'microsoft/do-client'
-      action: 'create'
-      target: '$(Build.SourceVersion)'
-      tagSource: 'userSpecifiedTag'
-      tag: 'ms-do-client-$(release.version)'
-      title: '$(release.title) $(release.version)'
-      changeLogCompareToRelease: 'lastNonDraftReleaseByTag'
-      changeLogCompareToReleaseTag: 'ms-do-client-$(release.previous_version)'
-      changeLogType: 'commitBased'
\ No newline at end of file
+- stage: release
+  condition: and(succeeded(), eq('${{parameters.SkipPublishing}}', false))
```

Comment:
>condition [](start = 2, length = 9)

When SkipPublishing is true, consider listing out, just to the pipeline log output, the artifacts that will be included in the release and things like tag and title. Is there a way to output the change log as well?

Reply:
I see, so a true dry-run. Let me see

---
In reply to: [586662450](https://github.com/microsoft/do-client/pull/7#discussion_r586662450) [](ancestors = 586662450)

Reply:
We can definitely print the artifacts to be published, release name and tag/title, but not a good way to output change log.

One hacky way of achieving the above and to print out the change log would be to publish a release, then immediately delete the release in a subsequent task if SkipPublishing is set to true.

---
In reply to: [586685709](https://github.com/microsoft/do-client/pull/7#discussion_r586685709) [](ancestors = 586685709,586662450)

Reply:
Ok, in that case, print everything except the change log. I don't want to actually publish to github because (I think) it might trigger emails if someone is following our repo.

---
In reply to: [589764290](https://github.com/microsoft/do-client/pull/7#discussion_r589764290) [](ancestors = 589764290,586685709,586662450)

Reply:
I've added a task under release to do this, and made the actual github release task the conditional one.

Do note that all that information is available anyways through the azure pipeline display -> all variables can be found as well as contents to be published, but the added task should make it explicit.

---
In reply to: [589767276](https://github.com/microsoft/do-client/pull/7#discussion_r589767276) [](ancestors = 589767276,589764290,586685709,586662450)